documentation/docs/05-loading.md

```diff
@@ -1,157 +0,0 @@
----
-title: Loading
----
-
-A component that defines a page or a layout can export a `load` function that runs before the component is created. This function runs both during server-side rendering and in the client, and allows you to fetch and manipulate data before the page is rendered, thus preventing loading spinners.
-
-If the data for a page comes from its endpoint, you may not need a `load` function. It's useful when you need more flexibility, for example loading data from an external API, which might look like this:
-
-```html
-/// file: src/routes/blog/[slug].svelte
-<script context="module">
-	/** @type {import('./__types/[slug]').Load} */
-	export async function load({ params, fetch, session, stuff }) {
-		const url = `https://cms.example.com/article/${params.slug}.json`;
-		const response = await fetch(url);
-
-		return {
-			status: response.status,
-			props: {
-				article: response.ok && (await response.json())
-			}
-		};
-	}
-</script>
-```
-
-> Note the `<script context="module">` — this is necessary because `load` runs before the component is rendered. Code that is per-component instance should go into a second `<script>` tag.
-
-As with [endpoints](/docs/routing#endpoints), pages can import [generated types](/docs/types#generated-types) — the `./[slug]` in the example above — to ensure that `params` are correctly typed.
-
-`load` is similar to `getStaticProps` or `getServerSideProps` in Next.js, except that `load` runs on both the server and the client. In the example above, if a user clicks on a link to this page the data will be fetched from `cms.example.com` without going via our server.
-
-SvelteKit's `load` receives an implementation of `fetch`, which has the following special properties:
-
-- it has access to cookies on the server
-- it can make requests against the app's own endpoints without issuing an HTTP call
-- it makes a copy of the response when you use it, and then sends it embedded in the initial page load for hydration
-
-`load` only applies to [page](/docs/routing#pages) and [layout](/docs/layouts) components (not components they import), and runs on both the server and in the browser with the default rendering options.
-
-> Code called inside `load` blocks:
->
-> - should use the SvelteKit-provided [`fetch`](/docs/loading#input-fetch) wrapper rather than using the native `fetch`
-> - should not reference `window`, `document`, or any browser-specific objects
-> - should not directly reference any API keys or secrets, which will be exposed to the client, but instead call an endpoint that uses any required secrets
-
-It is recommended that you not store per-request state in global variables, but instead use them only for cross-cutting concerns such as caching and holding database connections.
-
-> Mutating any shared state on the server will affect all clients, not just the current one.
```

Comment:
This is an important gotcha that some people are not aware of (as seen in related discussions). We should have this somewhere in the docs.

Reply:
> This is an important gotcha that some people are not aware of (as seen in related discussions). We should have this somewhere in the docs.

Agree with you. E.g:  I didn't know that, when I switched from SPA to SSR.

Reply:
I agree with keeping this: it comes up a lot of times with Serverless platforms like Cloudflare Workers where other requests are handled during async tasks, so storing request-specific things in global state can be disastrous.

Reply:
yeah, this was a casualty of restructuring this section of the docs to accommodate new API — should be reinstated somewhere

Reply:
added a note in c1e20860090554dcf947206be0b5bc1735932d99, will mark this resolved